### ISSUE TITLE
[Bug] Start-DbaAgentJob verbose output "Cannot convert value" error for job step

### ISSUE THREAD

**Issue by wsuhoey** [sqlcollaborative/dbatools#7444]
#### Host used

- [ ] powershell.exe
- [ ] ISE
- [ ] VS Code
- [X] Other (please specify) pwsh.exe

#### Errors Received

```
InvalidArgument: C:\Users\myusername\Documents\PowerShell\Modules\dbatools\1.0.153\allcommands.ps1:71864
 Line |
71864 |              [int]$currentStepId = $currentstep.Split('()')[0].Trim()
      |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      | Cannot convert value "7 (My Job Step Name)" to type "System.Int32". Error: "Input string was
      | not in a correct format."
```

```
PSMessageDetails      :
Exception             : System.Management.Automation.RuntimeException: Cannot convert value "7 (My Job Step Name
                        )" to type "System.Int32". Error: "Input string was not in a correct format."
                         ---> System.Management.Automation.PSInvalidCastException: Cannot convert value "7 (My Job
                        Step Name)" to type "System.Int32". Error: "Input string was not in a correct format."
                         ---> System.FormatException: Input string was not in a correct format.
                           at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
                           at System.Number.ParseInt32(ReadOnlySpan`1 value, NumberStyles styles, NumberFormatInfo
                        info)
                           at System.ComponentModel.Int32Converter.FromString(String value, NumberFormatInfo
                        formatInfo)
                           at System.ComponentModel.BaseNumberConverter.ConvertFrom(ITypeDescriptorContext context,
                        CultureInfo culture, Object value)
                           --- End of inner exception stack trace ---
                           at System.Management.Automation.LanguagePrimitives.ConvertStringToInteger(Object
                        valueToConvert, Type resultType, Boolean recursion, PSObject originalValueToConvert,
                        IFormatProvider formatProvider, TypeTable backupTable)
                           at CallSite.Target(Closure , CallSite , Object )
                           at <ScriptBlock>(Closure , FunctionContext )
                           --- End of inner exception stack trace ---
TargetObject          :
CategoryInfo          : InvalidArgument: (:) [], RuntimeException
FullyQualifiedErrorId : InvalidCastFromStringToInteger
ErrorDetails          :
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at <ScriptBlock>, C:\Users\myusername\Documents\PowerShell\Modules\dbatools\1.0.153\allcommands.ps1: line 71864
                        at Start-DbaAgentJob<Process>, C:\Users\myusername\Documents\PowerShell\Modules\dbatools\1.0.153\allcommands.ps1: line 71959
                        at <ScriptBlock>, C:\Users\myusername\Documents\PowerShell\Data
                        Refresh Delay reload data Prod.ps1: line 90
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}
```

#### Steps to Reproduce

`Start-DbaAgentJob -SqlInstance $server -SqlCredential $creds -Job $jobname -Wait -WaitPeriod 60 -Verbose`

#### Expected Behavior

Display proper job step.

#### Actual Behavior

Error output, and blank or empty variable outputs no job step name.
```
VERBOSE: [12:28:50][Start-DbaAgentJob] Server: MyServer - My Agent Job is Executing, currently on Job Step 'My Job Step Name' (7 of 40), and has tried 0 of 0 retry attempts
```

see /functions/Start-DbaAgentJob.ps1 `line 140`:
```
            [int]$currentStepId = $currentstep.Split('()')[0].Trim()
```

### Environmental information

🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
Please provide the output of the below script

```powershell
[pscustomobject]@{
'PowerShell Version' = $PSVersionTable.PSVersion.ToString()
'dbatools latest installed' = (Get-InstalledModule -Name dbatools).Version
'Culture of OS' = (Get-Culture)
} | fl -force
```
PowerShell Version        : 7.1.3
dbatools latest installed : 1.0.153
Culture of OS             : en-US

#### SQL Server:

```sql
Microsoft SQL Server 2017 (RTM-CU22-GDR) (KB4583457) - 14.0.3370.1 (X64) 
	Nov  6 2020 18:19:52 
	Copyright (C) 2017 Microsoft Corporation
	Standard Edition (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) (Hypervisor)

and

Microsoft SQL Server 2019 (RTM-CU8-GDR) (KB4583459) - 15.0.4083.2 (X64) 
	Nov  2 2020 18:35:09 
	Copyright (C) 2019 Microsoft Corporation
	Standard Edition (64-bit) on Windows Server 2019 Standard 10.0 <X64> (Build 17763: ) (Hypervisor)

```

```sql
us_english
```

**Comment by niphlod** [comment]
ouch. there's a nicer way but the interface is wonky . will work on that if you can wait a few days.

**Comment by niphlod** [comment]
@wsuhoey : would you give this a whirl ? https://github.com/sqlcollaborative/dbatools/pull/7445

**Comment by wsuhoey** [comment]
I realize I *should* know this, but how do I test the updated code? Do I edit the code in Start-DbaAgentJob.ps1 or do I find the section in allcommands.ps1? I don't normally do this :)

**Comment by wsuhoey** [comment]
Confirmed fixed in `1.0.158`
